[25.1] Don't create extra watchers when switching tabs

natefoo opened the pull request:
Suggested fix from @mvdbeek. This is causing way more polling of `current_history_json` than there should be.

I'm applying directly to the usegalaxy_25.1 so we can keep this PR open for a bit if we have more to add later.

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

natefoo committed:
Don't create extra watchers when switching tabs

davelopez commented:
Thank you for catching that @natefoo!
I'll take a look at the unit test failure and fix it :+1:

davelopez committed:
Updates watcher tests for visibility change polling

Clarifies test expectations to verify that switching to a short polling interval on app visibility does not duplicate the resource watcher or trigger extra handler calls.
